spring/boot-autoconfigure/src/main/java/com/linecorp/armeria/spring/ArmeriaBeanPostProcessor.java

```diff
@@ -0,0 +1,214 @@
+                    }
+                    currElements.add(new LocalArmeriaPortElement(field, field, null));
+                } else if (field.isAnnotationPresent(LocalArmeriaPorts.class)) {
+                    if (Modifier.isStatic(field.getModifiers())) {
+                        throw new IllegalStateException(
+                                "LocalArmeriaPorts annotations are not supported on static fields");
+                    }
+                    currElements.add(new LocalArmeriaPortsElement(field, null));
+                }
+            });
+
+            ReflectionUtils.doWithLocalMethods(targetClass, method -> {
+                final Method bridgedMethod = BridgeMethodResolver.findBridgedMethod(method);
+                if (!BridgeMethodResolver.isVisibilityBridgeMethodPair(method, bridgedMethod)) {
+                    return;
+                }
+                if (bridgedMethod.isAnnotationPresent(LocalArmeriaPort.class)) {
+                    if (Modifier.isStatic(method.getModifiers())) {
+                        throw new IllegalStateException(
+                                "LocalArmeriaPort annotations are not supported on static methods");
+                    }
+                    final PropertyDescriptor pd = BeanUtils.findPropertyForMethod(bridgedMethod, clazz);
+                    currElements.add(new LocalArmeriaPortElement(method, bridgedMethod, pd));
+                } else if (bridgedMethod.isAnnotationPresent(LocalArmeriaPorts.class)) {
+                    if (Modifier.isStatic(method.getModifiers())) {
+                        throw new IllegalStateException(
+                                "LocalArmeriaPorts annotations are not supported on static methods");
+                    }
+                    final PropertyDescriptor pd = BeanUtils.findPropertyForMethod(bridgedMethod, clazz);
+                    currElements.add(new LocalArmeriaPortsElement(method, pd));
+                }
+            });
+
+            elements.addAll(0, currElements);
+            targetClass = targetClass.getSuperclass();
+        }
+        while (targetClass != null && targetClass != Object.class);
+
+        return new InjectionMetadata(clazz, elements);
+    }
+
+    /**
+     * Class representing injection information about an annotated field
+     * or setter method, supporting the {@link LocalArmeriaPort}.
+     */
+    private final class LocalArmeriaPortElement extends InjectionMetadata.InjectedElement {
+
+        private final int port;
+
+        private LocalArmeriaPortElement(Member member, AnnotatedElement ae, @Nullable PropertyDescriptor pd) {
+            super(member, pd);
+            final LocalArmeriaPort localArmeriaPort = ae.getAnnotation(LocalArmeriaPort.class);
+            final SessionProtocol protocol = localArmeriaPort.value();
+            final Server server = beanFactory.getBean(Server.class);
```

Comment:
Could we reuse this instance?

Reply:
Let's find a way to reuse it. In addition, it would be good to add a value to`Map`.

Reply:
Fixed 98f2e75